### PR TITLE
Expose device_information

### PR DESCRIPTION
**Pull request by ConradIrwin**
We'd like to be able to tell users not to use a software renderer, and to provide more information in the case of failures.

Fixes: kvark/blade#142

I have not yet implemented this on metal, but can do so if this approach seems sound to you.

### CODE REVIEW

**Review by kvark — COMMENTED** [review]
Generally speaking, exposing some information about the device is good and helpful.

However, I would very much like to squeeze this information to a form that is useful, as opposed to blindly exposing everything that Vulkan gives us. For example, Vulkan drivers aren't even in agreement about what is considered an integrated GPU and what not. This line is being blurred, and it's not really actionable by the user anyway.

What the user does need to know is if the adapter is emulated in software. That is definitely a strong distinction, where the line isn't blurred. So, we would ideally expose something like this:
```
struct DeviceInformation {
  name: String,
  is_software_emulated: bool,
}
```

**Comment by ConradIrwin** [comment]
@kvark :+1: can make that change.

I had a few people on the linux channel suggest it would be useful to return the driver info too to help people debug what's happening. Is that information obtainable a different way, or should I keep returning that?

**Comment by ConradIrwin** [comment]
@kvark updated to just return a bool. Kept all three strings, but can simplify if you'd prefer

File: blade-graphics/src/metal/mod.rs [hunk 1]
```diff
@@ -465,6 +465,10 @@ impl Context {
         }
     }
 
+    pub fn device_information(&self) -> &crate::DeviceInformation {
+        todo!();
```
Comment (kvark) [inline blade-graphics/src/metal/mod.rs:469]: Can we return a default implementation instead? Crashing isn't fun for this

**Review by kvark — COMMENTED** [review]
Sorry, one last thing

**Comment by ConradIrwin** [comment]
@kvark I'm keen to ship a new build to zed preview on Wednesday. I can pull this in, but is #144 something you feel comfortable with too?

**Comment by kvark** [comment]
@ConradIrwin I don't want to merge #144 until we confirm that it helps people.

**Pull request merged**